core/dbt/artifacts/resources/v1/semantic_model.py

```diff
@@ -42,7 +42,7 @@ class NodeRelation(dbtClassMixin):
     alias: str
     schema_name: str  # TODO: Could this be called simply "schema" so we could reuse StateRelation?
     database: Optional[str] = None
-    relation_name: Optional[str] = None
```

Comment:
Are we using this field anywhere?
Otherwise, might be OK to drop similar to https://github.com/dbt-labs/dbt-core/pull/9794

Reply:
I will look up but @QMalcolm  if you know anything about it

Reply:
It's used in a couple places in [Metricflow](https://github.com/dbt-labs/metricflow):
- [convert_semantic_model.py::create_sql_source_data_set](https://github.com/dbt-labs/metricflow/blob/8cc8eb79c7e2c7456bb51318443e906b1108690d/metricflow/dataset/convert_semantic_model.py#L496)
- [data_warehouse_model_validator.py::gen_semantic_model_tasks](https://github.com/dbt-labs/metricflow/blob/8cc8eb79c7e2c7456bb51318443e906b1108690d/metricflow/model/data_warehouse_model_validator.py#L142)

Thus it is important for us to power it.